File: java/json-transform/src/test/java/co/nlighten/jsontransform/functions/TransformerFunctionMatchTest.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.BaseTest;
import org.junit.jupiter.api.Test;

/**
 * @see TransformerFunctionMatch
 */
public class TransformerFunctionMatchTest extends BaseTest {
    @Test
    void testInlineFunctionMatch() {
        var input = "hello";
        assertTransformation(input, "$$match([le]):$", "e");
        assertTransformation(input, "$$match([le]+):$", "ell");
        assertTransformation(input, "$$match(hell):$", "hell");
        assertTransformation(input, "$$match(hello$):$", "hello");
        assertTransformation(input, "$$match(hell$):$", null);
    }
    @Test
    void inlineGroup() {
        assertTransformation("world", "$$match('w(\\\\w+)d',1):$", "orl");
    }
}

File: javascript/json-transform/src/__tests__/functions/TransformerFunctionMatch.test.ts
import { describe, test } from "vitest";
import { assertTransformation } from "../BaseTransformationTest";

describe("TransformerFunctionMatch", () => {
  test("inline", async () => {
    const input = "hello";
    await assertTransformation(input, "$$match([le]):$", "e");
    await assertTransformation(input, "$$match([le]+):$", "ell");
    await assertTransformation(input, "$$match(hell):$", "hell");
    await assertTransformation(input, "$$match(hello$):$", "hello");
    await assertTransformation(input, "$$match(hell$):$", null);
  });

  test("inlineGroup", async () => {
    await assertTransformation("world", "$$match('w(\\\\w+)d',1):$", "orl");
  });
});
